src/ck3/data/wars.rs

```diff
@@ -0,0 +1,94 @@
+use std::path::PathBuf;
+
+use crate::block::Block;
+use crate::everything::Everything;
+use crate::fileset::{FileEntry, FileHandler};
+use crate::item::Item;
+use crate::pdxfile::PdxFile;
+use crate::report::{err, warn, ErrorKey};
+use crate::token::Token;
+use crate::validator::Validator;
+
+#[derive(Clone, Debug, Default)]
+pub struct Wars {
+    wars: Vec<War>,
+}
+
+impl Wars {
+    fn load_item(&mut self, key: Token, block: Block) {
+        if key.is("war") {
+            self.wars.push(War::new(block));
+        } else {
+            let msg = format!("unexpected key {key}, expected only `war`");
+            err(ErrorKey::History).msg(msg).loc(key).push();
+        }
+    }
+
+    pub fn validate(&self, data: &Everything) {
+        for item in &self.wars {
+            item.validate(data);
+        }
+    }
+}
+
+impl FileHandler<Block> for Wars {
+    fn subpath(&self) -> PathBuf {
+        PathBuf::from("history/wars")
+    }
+
+    fn load_file(&self, entry: &FileEntry) -> Option<Block> {
+        if !entry.filename().to_string_lossy().ends_with(".txt") {
+            return None;
+        }
+
+        PdxFile::read_optional_bom(entry)
+    }
+
+    fn handle_file(&mut self, _entry: &FileEntry, mut block: Block) {
+        for (key, block) in block.drain_definitions_warn() {
+            self.load_item(key, block);
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct War {
+    block: Block,
+}
+
+impl War {
+    pub fn new(block: Block) -> Self {
+        Self { block }
+    }
+
+    fn validate(&self, data: &Everything) {
+        let mut vd = Validator::new(&self.block, data);
+
+        vd.req_field("start_date");
+        vd.req_field("end_date");
```

Comment:
Other fields (`targeted_titles`, `casus_belli`, `attackers`, `defenders`) would likely be required as well. Best to verify